cluster.go

```diff
@@ -205,6 +205,15 @@ func (c *clusterClient) _refresh() (err error) {
 	slots := [16384]conn{}
 	for master, g := range groups {
 		cc := conns[master]
+		if c.opt.ReplicaOnly {
+			if replicaAddr := g.pickReplica(); replicaAddr != "" {
+				cc = conns[replicaAddr]
+				result := cc.Do(context.Background(), cmds.ReadOnlyCmd)
```

Comment:
Hi @NeoHuang,

Thank you for implementing the `ReplicaOnly` for the cluster client.

Your implementation generally looks good to me. However, setting the `READONLY` state right here is not a good idea. It will not only make this refresh function slow but also make it hard to finish its job.

I think it will be better to pipeline the `READONLY` right after the `SELECT` command.

Here for RESP3:
https://github.com/redis/rueidis/blob/6556f2fa4d07c416b3029f04419ef77067f09b8a/pipe.go#L173-L175

And here for RESP2:
https://github.com/redis/rueidis/blob/6556f2fa4d07c416b3029f04419ef77067f09b8a/pipe.go#L247-L249

We can probably rely on the `option.ReplicaOnly` flag to pipeline the `READONLY` command after these two blocks if the pipe is managed under a cluster client.

Reply:
Hi @rueian , thanks for the feedback and I think it's a good idea. I'll make that change

Reply:
actually if we call it here in newPipe, it would panic if the redis doesn't have the cluster mode enabled. but we don't seem to have any context that whether cluster mode is supported or not here as we always try cluster mode first? 
any idea how we can get around?

Reply:
Was the panic you mentioned caused by the `return nil, err` here?
https://github.com/redis/rueidis/blob/19e115b2d48d3e434542cd22dcd9b15f873a3285/pipe.go#L217-L218

We probably need to intercept that case and return `p, err` instead.

Reply:
it's panic here https://github.com/redis/rueidis/blob/19e115b2d48d3e434542cd22dcd9b15f873a3285/rueidis.go#L306

the client is nil and not checked in this line. and yeah, the nil client was returned by your above code, but I am not sure if it's a good idea to intercept that specific case. I think I would just return error after this line

https://github.com/redis/rueidis/blob/19e115b2d48d3e434542cd22dcd9b15f873a3285/rueidis.go#L303
like this
```go
	if client, err = newClusterClient(&option, makeConn); err != nil {
		if client == nil || client == (*clusterClient)(nil) {
			return nil, err
		}
```

reason is that setting ReplicaOnly on single client is not expected anyway, I would rather throw this error instead of hiding it. also it would add more graceful handling of nil client cases for other scenarios. WDYT? :)

Reply:
Because if the target server belongs to a redis cluster, the `READONLY` command will succeed, and if the target server doesn't belong to a cluster, the `READONLY` command will have no effect.

Reply:
So, we can safely pipeline the `READONLY` command if `option.ReplicaOnly` is set without checking its response.